src/Plugins/SimplnxCore/src/SimplnxCore/Filters/Algorithms/WriteLAMMPSFile.cpp

```diff
@@ -0,0 +1,154 @@
+  if(createDirectoriesResult.invalid())
+  {
+    return createDirectoriesResult;
+  }
+
+  std::ofstream file = std::ofstream(m_InputValues->OutputFile, std::ios_base::out | std::ios_base::binary);
+  if(!file.is_open())
+  {
+    return MakeErrorResult(-77450, fmt::format("Error creating and opening output file at path: {}", m_InputValues->OutputFile.string()));
+  }
+
+  const auto& vertexGeom = m_DataStructure.getDataAs<VertexGeom>(m_InputValues->VertexGeomPath);
+  const AbstractDataStore<VertexGeom::SharedVertexList::value_type>& verts = vertexGeom->getVertices()->getDataStoreRef();
+
+  const Int32AbstractDataStore& atomLabels = m_DataStructure.getDataAs<Int32Array>(m_InputValues->AtomLabelsPath)->getDataStoreRef();
+
+  MessageHelper messageHelper(m_MessageHandler);
+
+  messageHelper.sendMessage("Finding Max Atom Label...");
+  int32 atomTypes = 0;
+  for(usize i = 0; i < atomLabels.getNumberOfTuples(); i++)
+  {
+    if(m_ShouldCancel)
+    {
+      return {};
+    }
+    int32 atomLabel = atomLabels.getValue(i);
+    if(atomLabel > atomTypes)
+    {
+      atomTypes = atomLabel;
+    }
+  }
+
+  float xMin = 1000000000.0;
+  float xMax = 0.0;
+  float yMin = 1000000000.0;
+  float yMax = 0.0;
+  float zMin = 1000000000.0;
+  float zMax = 0.0;
+  int dummy = 0;
+
+  messageHelper.sendMessage("Finding Min/Max Vertices...");
+  for(usize i = 0; i < verts.getNumberOfTuples(); i++)
+  {
+    if(m_ShouldCancel)
+    {
+      return {};
+    }
+    std::array<float32, 3> pos = {verts.getValue((i * 3) + 0), verts.getValue((i * 3) + 1), verts.getValue((i * 3) + 2)};
+    if(pos[0] < xMin)
+    {
+      xMin = pos[0];
+    }
+    if(pos[0] > xMax)
+    {
+      xMax = pos[0];
+    }
+    if(pos[1] < yMin)
+    {
+      yMin = pos[1];
+    }
+    if(pos[1] > yMax)
+    {
+      yMax = pos[1];
+    }
+    if(pos[2] < zMin)
+    {
+      zMin = pos[2];
+    }
+    if(pos[2] > zMax)
+    {
+      zMax = pos[2];
+    }
+  }
+
+  messageHelper.sendMessage("Writing File Metadata...");
+  file << "LAMMPS data file\n";
+  file << "\n";
+  file << fmt::format("{} atoms\n", static_cast<long long int>(verts.getNumberOfTuples()));
+  file << "\n";
+  file << fmt::format("{} atom types\n", static_cast<long long int>(atomTypes));
+  file << "\n";
+  file << fmt::format("{:f} {:f} xlo xhi\n", xMin, xMax);
+  file << fmt::format("{:f} {:f} ylo yhi\n", yMin, yMax);
+  file << fmt::format("{:f} {:f} zlo zhi\n", zMin, zMax);
+  file << "\n";
+  file << "Atoms\n";
+  file << "\n";
+
+  messageHelper.sendMessage("Exporting Data...");
+  ThrottledMessenger throttledMessenger = messageHelper.createThrottledMessenger();
+  // Write the Atom positions (Vertices)
+  usize numVerts = verts.getNumberOfTuples();
+  for(usize i = 0; i < numVerts; i++)
+  {
+    if(m_ShouldCancel)
+    {
+      return {};
+    }
+    throttledMessenger.sendThrottledMessage([&]() { return fmt::format("Exporting Data {:.2f}% completed", CalculatePercentComplete(i, numVerts)); });
```

Comment:
How expensive is this call. Maybe we should just check every "Nth" vert. 

Also I think we should build up a buffer or set the buffer on the "file" object to be like 4MB at the minimum.

Create a geometry that is 1000 x 1000 x 1000 and write the file and measure how long it takes. We should try to maximize that through put best we can.